sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -92,6 +107,20 @@
  * PCollection<String> lines = filenames.apply(TextIO.readAll());
  * }</pre>
  *
+ * <p>Example 3: streaming new files matching a filepattern.
+ *
+ * <pre>{@code
+ * Pipeline p = ...;
+ *
+ * PCollection<String> lines = p.apply(TextIO.read()
+ *     .from("/local/path/to/files/*")
```

Comment:
Imagine a case where a new directory is generated every day, and you then want to start watching for new files in that directory. Can we model this?